src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -535,12 +540,11 @@ grpc_chttp2_transport::grpc_chttp2_transport(
 static void destroy_transport_locked(void* tp, grpc_error_handle /*error*/) {
   grpc_chttp2_transport* t = static_cast<grpc_chttp2_transport*>(tp);
   t->destroying = 1;
-  grpc_resource_user_shutdown(t->resource_user);
-  grpc_resource_user_unref(t->resource_user);
   close_transport_locked(
       t, grpc_error_set_int(
              GRPC_ERROR_CREATE_FROM_STATIC_STRING("Transport destroyed"),
              GRPC_ERROR_INT_OCCURRED_DURING_WRITE, t->write_state));
+  [](grpc_core::MemoryOwner) {}(std::move(t->memory_owner));
```

Comment:
Does this accomplish anything without calling or storing the lambda? And even then if it was executed, what does it accomplish without returning or assigning the result of `std::move`?

Reply:
It calls the move constructor without introducing another name... effectively ensuring the owner is cleared out afterwards... but this is too obscure, going to do something cleaner.

Reply:
My mistake, the lambda is indeed being called here. Agreed, that maybe something more straightforward would be better.